pkg/piecestore/pstore.go

```diff
@@ -32,73 +32,85 @@ func PathByHash(hash, dir string) (string, error) {
 	return path.Join(dir, folder1, folder2, fileName), nil
 }
 
-// Store -- Store data into piece store
-// 	hash 					(string)		Hash of the data to be stored
-// 	r 			  		(io.Reader)	File/Stream that contains the contents of the data to be stored
-// 	length 				(length)		Size of the data to be stored
-// 	psFileOffset 	(offset)  	Offset of the data that you are writing. Useful for multiple connections to split the data transfer
-// 	dir 					(string)		pstore directory containing all other data stored
-// 	returns 			(error) 		error if failed and nil if successful
-func Store(hash string, r io.Reader, length int64, psFileOffset int64, dir string) (int64, error) {
+// StoreWriter -- Store data into piece store in multiple writes
+// 	hash 					(string)								Hash of the data to be stored
+// 	dir 					(string)								pstore directory containing all other data stored
+// 	returns 			(*fpiece.Chunk, error) 	error if failed and nil if successful
+func StoreWriter(hash string, length int64, psFileOffset int64, dir string) (*fpiece.Chunk, error) {
 	if psFileOffset < 0 {
-		return 0, ArgError.New("Offset is less than 0. Must be greater than or equal to 0")
-	}
-	if length < 0 {
-		return 0, ArgError.New("Length is less than 0. Must be greater than or equal to 0")
+		return nil, ArgError.New("Offset is less than 0. Must be greater than or equal to 0")
 	}
 	if dir == "" {
-		return 0, ArgError.New("No path provided")
+		return nil, ArgError.New("No path provided")
 	}
 
 	dataPath, err := PathByHash(hash, dir)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
 
 	// Create directory path on file system
 	if err = os.MkdirAll(filepath.Dir(dataPath), 0700); err != nil {
-		return 0, err
+		return nil, err
 	}
 
 	// Create File on file system
 	dataFile, err := os.OpenFile(dataPath, os.O_RDWR|os.O_CREATE, 0755)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
 
-	dataFileChunk := fpiece.NewChunk(dataFile, psFileOffset, length)
+	return fpiece.NewChunk(dataFile, psFileOffset, length), nil
+}
+
+// Store -- Store data into piece store
+// 	hash 					(string)				Hash of the data to be stored
+// 	r 			  		(io.Reader)			File/Stream that contains the contents of the data to be stored
+// 	length 				(length)				Size of the data to be stored
+// 	psFileOffset 	(offset)  			Offset of the data that you are writing. Useful for multiple connections to split the data transfer
+// 	dir 					(string)				pstore directory containing all other data stored
+// 	returns 			(int64, error) 	error if failed and nil if successful
+func Store(hash string, r io.Reader, length int64, psFileOffset int64, dir string) (int64, error) {
```

Comment:
StoreWriter does more things and works in more places than Store. I'm a big proponent of having the least amount of code in the codebase possible that achieves our goals. less code to maintain and support later. since StoreWriter totally solves the issue of grpc system call overhead, can we just use StoreWriter entirely and get rid of Store?

Reply:
I'm fine with this. I need to redo the tests anyway 😅

Reply:
ha sorry